### PR TITLE
Add a separate doc for versioning info

### PR DESCRIPTION
**Pull request by thecoop**
New doc containing versioning info

### CODE REVIEW

File: VERSIONING.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+`ESRestTestCase.clusterHasFeature(feature)`
+
+In YAML Rest tests, conditions can be defined in the `requires` or `skip` sections
+that use cluster features; see [here](https://github.com/elastic/elasticsearch/blob/main/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc#skipping-tests) for more information.
+
+To aid with backwards compatibility tests, the test framework adds synthetic features
+for each previously released Elasticsearch version, of the form `gte_v{VERSION}`
+(for example `gte_v8.14.2`).
+This can be used to add conditions based on previous releases. It cannot be used
+to check the current snapshot version; real features or capabilities should be
```
Comment (mosche) [inline VERSIONING.md:243]: > It cannot be used to check the current snapshot version

Should we be more vocal about this? E.g. "However, it is forbidden to check for the current snapshot version."

**Review by mosche — APPROVED** [review]
Great write up 💯 
LGTM, just a small optional suggestion / question.

File: VERSIONING.md (hunk 1 above)
Comment (thecoop) [reply VERSIONING.md:243]: That might be a bit too extreme... I've added an emphasis to the _cannot_

**Review by rjernst — COMMENTED** [review]
Doc looks good. Small nit: can we put this under `docs/internal/`? The root of the project is littered with too many files. I think we should start moving them and then reference/link from README or CONTRIBUTING.

**Pull request merged**